src/services/route.rs

```diff
@@ -37,6 +69,35 @@ pub struct RouteInfo {
     pub fragment: Option<String>
 }
 
+impl Iterator for RouteInfo {
```

Comment:
I added an iterator to RouteInfo. I would love it something like this could be used to only expose relevant path information to particular components.

So the main Model would get the first segment, and then its child would get the next, etc, and would only expose the query and fragment to the leafs in tree of components.

`tune_from_part()` would take a RouteSection and use that to determine its child instead of a &str.